pkg/apis/core/v1/helper/helpers.go

```diff
@@ -200,30 +200,30 @@ func containsAccessMode(modes []v1.PersistentVolumeAccessMode, mode v1.Persisten
 	return false
 }
 
-// NodeSelectorRequirementsAsSelector converts the []NodeSelectorRequirement api type into a struct that implements
+// NumericAwareSelectorRequirementsAsSelector converts the []NumericAwareSelectorRequirement api type into a struct that implements
 // labels.Selector.
-func NodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement) (labels.Selector, error) {
+func NumericAwareSelectorRequirementsAsSelector(nsm []v1.NumericAwareSelectorRequirement) (labels.Selector, error) {
 	if len(nsm) == 0 {
 		return labels.Nothing(), nil
 	}
 	selector := labels.NewSelector()
 	for _, expr := range nsm {
 		var op selection.Operator
 		switch expr.Operator {
-		case v1.NodeSelectorOpIn:
+		case v1.LabelSelectorOpIn:
```

Comment:
If operator/customized-controller use `v1.NodeSelectorOpIn`, they have to update the code after upgrade client-go, similar to what we have done in daemonset_controller_test in this PR.

Reply:
This seems to be no problem, the value of `NodeSelectorOpIn` is "In" string, which is consistent with the current `LabelSelectorOpIn`. In other words, even if you don't upgrade client-go (using the version before the PR), the operator is still "In". 
Don't know if my understanding is correct or answer your question?

Reply:
I will take a deep look at this. :)

Reply:
and similar comments to other field/struct :)

Reply:
I did some experimentation and testing, and I feel that we are in a dilemma. The best way to satisfy compatibility and implement current features is to implement `PodSelector` independently. But from a code point of view, everyone tends to abstract `NodeSelector` and `PodSelector`, but this must be done by changing the API, that is, the go struct must be changed. It seems that we can't find a perfect solution.

How to implement this part of the code, maybe I need some advice and help.

I remember @k82cn  once mentioned the solution:

> Golang support constructing nested struct without name, but it dependent on the coding style

Can you clarify this plan?

Maybe @bsalamat @Huang-Wei and @misterikkit has some other good suggestions?

Reply:
@k82cn Please help me understand what the problem is. These operators are string literals. Are we concerned that someone may have a "typed" operators in their existing config that would break after upgrading?

Reply:
>  Or is it only about custom controllers that use existing API code?

Yes, it's only about custom controllers/operators; if using http call, that's fine.